From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Siddharth Sharma <siddharth@redhat.com>
Date: Thu, 4 Jul 2024 01:18:47 -0400
Subject: [PATCH] 6691: Fix null pointer exception in createGot method

Add a null check before accessing 'block' in the createGot method to prevent a potential NullPointerException. The check ensures that the method returns early if 'block' is null, which can occur if MemoryBlock creation fails or is not initialized. This change improves robustness of the method by handling unexpected null references gracefully.
---
 .../bin/format/elf/relocation/X86_64_ElfRelocationContext.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Ghidra/Processors/x86/src/main/java/ghidra/app/util/bin/format/elf/relocation/X86_64_ElfRelocationContext.java b/Ghidra/Processors/x86/src/main/java/ghidra/app/util/bin/format/elf/relocation/X86_64_ElfRelocationContext.java
index eae466cde5..6bc4074966 100644
--- a/Ghidra/Processors/x86/src/main/java/ghidra/app/util/bin/format/elf/relocation/X86_64_ElfRelocationContext.java
+++ b/Ghidra/Processors/x86/src/main/java/ghidra/app/util/bin/format/elf/relocation/X86_64_ElfRelocationContext.java
@@ -260,6 +260,9 @@ class X86_64_ElfRelocationContext extends ElfRelocationContext<X86_64_ElfRelocat
 				"NOTE: This block is artificial and allows ELF Relocations to work correctly",
 				"Elf Loader", true, false, false, loadHelper.getLog());
 
+			if (block == null) {
+				return;
+			}
 			// Mark block as an artificial fabrication
 			block.setArtificial(true);
 
-- 
2.45.1

